Switch TR-55 Water Quality to Bullet Charts

rajadain opened the pull request:
## Overview

Previously the water quality data was represented as a horizontal bar chart, with one bar for each type of observation. Unfortunately, the values of the three observations was quite disparate in scale, making the smallest bar insignificantly small.

To help bring each value to first class citizenship, we switch to using three bullet charts. Each chart is scaled to suit the value thereof, and thus shows real, actionable information.

Connects #2165 

### Demo

![image](https://user-images.githubusercontent.com/1430060/32628923-7a931faa-c565-11e7-8697-c2131670c8f1.png)

Tagging @jfrankl and @ajrobbins for visual review.

### Notes

The chart labels in the mock-up are left-aligned, but I couldn't figure out how to left-align the text easily here. Is the right alignment (default design for bullet charts) good enough?

The actual value isn't shown on the chart, but in the tooltip, which looks like this:

![image](https://user-images.githubusercontent.com/1430060/32629397-0d43da28-c567-11e7-9ce1-5ee23312628b.png)

But enabling that tooltip also enables this one:

![image](https://user-images.githubusercontent.com/1430060/32629420-22bcf628-c567-11e7-92a0-23a823f18cf9.png)

and I couldn't figure out a way to have one and not the other.

## Testing Instructions

 * Check out this branch and `bundle`
 * Go to [:8000/](http://localhost:8000/) and create or open a TR-55 project
 * Go to the Water Quality tab. Ensure the new chart is there, and the values correspond to the table.

ajrobbins commented:
Looks good to me visually! How are we getting the max values? It is a little weird to have that tooltip, but doesn't bother me too much.

rajadain commented:
Adding a GIF of the tooltip to better communicate:

![2017-11-09 17 09 07](https://user-images.githubusercontent.com/1430060/32632231-d0dcc702-c570-11e7-9d71-3a7a3b308701.gif)

kellyi commented:
Taking a look at this!

kellyi commented:
This looks great!

![screen shot 2017-11-09 at 5 20 40 pm](https://user-images.githubusercontent.com/4165523/32632635-57bd6c3a-c572-11e7-874d-11dfcdcda458.png)

kellyi commented:
> How are we getting the max values?

Just to clarify: we are calculating the max value as the top of the current value's order of magnitude here?

https://github.com/WikiWatershed/model-my-watershed/pull/2503/files#diff-3e33aa79d87dd300905a2a0b6c714e56R611

That seems reasonable to me: it'd ensure the charts always show an appropriate ratio.

rajadain commented:
> we are calculating the max value as the top of the current value's order of magnitude 

Yes.

kellyi commented:
I'm in favor of leaving the chart title & subtitle left justified as is.

I was able to move them around manually in Chrome by adding an attribute like `dx="-3em"` to the `nv-title` and `nv-subtitle` classes, but I don't know whether nvd3 has an interface to do this automatically:

![screen shot 2017-11-09 at 5 34 47 pm](https://user-images.githubusercontent.com/4165523/32633267-d6722a6e-c574-11e7-98d8-b2613b71ee56.png)

![screen shot 2017-11-09 at 5 35 49 pm](https://user-images.githubusercontent.com/4165523/32633256-cf7627a6-c574-11e7-9c33-a32b84415ab7.png)

However, it'd be a chore to have to set those manually & verify correctness across browsers & screen sizes.

kellyi commented on src/mmw/js/src/core/chart.js:
```diff
@@ -596,9 +596,45 @@ function renderCompareMultibarChart(chartEl, name, label, colors, stacked, yMax,
+    }
+    while (data > 10) {
+        data /= 10;
+        adjustment += 1;
```
`data` looks not to be used again after these adjustments (I checked in the REPL and the `measures: [data]` is passed by value.

Consider making a small utility function which would take a value of whatever magnitude and return its base 10 exponent & adding tests for < 1, > 10, and single digit values.

ajrobbins commented:
I'm fine with it.

jfrankl commented:
LGTM!

rajadain commented:
Rebased on latest `develop`, extracted the range in magnitude logic to a utility function and added some tests (and discovered some failing boundary conditions, so good idea @kellyi!). Will merge when green. Thanks for all the reviews!
